src/services/arweave/SimpleArweaveDataProvider.ts

```diff
@@ -98,6 +101,58 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
     });
   }
 
+  async validateArweaveAddress(address: string): Promise<undefined | boolean> {
+    const targetAddress = new ArweaveTransactionID(address);
+
+    const txPromise = this._arweave.api
+      .get(`/tx/${targetAddress.toString()}`)
+      .then((res: any) => (res.status === 200 ? res : false));
+
+    const balancePromise = this._arweave.api
+      .get(`/wallet/${targetAddress.toString()}/balance`)
+      .then((res: any) => (res.status === 200 ? res.data > 0 : false));
+
+    const gqlPromise = this._arweave.api
+      .post(`/graphql`, transactionByOwnerQuery(targetAddress))
+      .then((res: any) =>
+        res.status === 200 ? res.data.data.transactions.edges : false,
+      );
+
+    const [isTransaction, balance, hasTransactions] = await Promise.all([
+      txPromise,
+      balancePromise,
+      gqlPromise,
+    ]);
+
+    if (isTransaction) {
+      const tags = tagsToObject(isTransaction.data.tags);
+      const isContract = Object.values(tags).includes('SmartWeaveContract');
+      if (isContract) {
+        throw new Error(
+          `Provided address (${targetAddress.toString()}) is a SmartWeave Contract.`,
+        );
+      }
+      throw new Error(
+        `Provided address (${targetAddress.toString()} is a transaction ID and not a Smartweave Contract or an address.`,
+      );
+    }
+    if (balance) {
+      return true;
+    }
+    if (!balance) {
+      // test address : ceN9pWPt4IdPWj6ujt_CCuOOHGLpKu0MMrpu9a0fJNM
+      // must be connected to a gateway that fetches l2 to perform this check
+      if (!hasTransactions || !hasTransactions.length) {
+        throw new Error(
+          `Unable to verify address - this may mean the provided address (${targetAddress.toString()}) exists and has no transactions.`,
```

Comment:
shorten this error. `Unable to verify this is an arweave address. Please double check before proceeding.` or something similar

Reply:
added